plugins/modules/web_infrastructure/jira.py

```diff
@@ -320,12 +371,33 @@ def edit(restbase, user, passwd, params):
     return ret
 
 
+def update(restbase, user, passwd, params):
+    data = {
+        "update": params['fields']
+    }
+    url = restbase + '/issue/' + params['issue']
+
+    ret = put(url, user, passwd, params['timeout'], data)
+
+    return ret
+
+
 def fetch(restbase, user, passwd, params):
     url = restbase + '/issue/' + params['issue']
     ret = get(url, user, passwd, params['timeout'])
     return ret
 
 
+def search(restbase, user, passwd, params):
+    url = restbase + '/search?jql=' + urllib.request.pathname2url(params['jql']) + '&maxResults=10'
+    if params['fields']:
+        fields = params['fields'].keys()
+        for f in fields:
+            url = url + '&fields=' + urllib.request.pathname2url(f)
```

Comment:
Something like that should work more efficient (just a suggestion)
```suggestion
        url = url +  '&fields=' + '&fields='.join([urllib.request.pathname2url(f) for f in fields])
```